### PR TITLE
feat(arc-313): Update space

### PR DESCRIPTION
**Pull request by bartnaessens**
- update basic space props: description, serviceDescription, color
- upload image file
- store file in S3
- Cron job to clean the upload dir
- delete file when 'image' is set empty
CleanShot 2022-03-24 at 16 18 57

### CODE REVIEW

File: src/modules/spaces/controllers/spaces.controller.ts [hunk 1]
```diff
@@ -34,11 +46,56 @@ export class SpacesController {
...
+			updateSpaceDto.image = await this.assetsService.upload(AssetFileType.SPACE_IMAGE, file);
+			if (space.image) {
+				// space already has an image: delete existing one
+				await this.assetsService.delete(space.image);
+			}
+		} else if (space.image && updateSpaceDto.image === '') {
+			// image is empty: delete current image
+			await this.assetsService.delete(space.image);
+		}
+		space = await this.spacesService.update(id, updateSpaceDto);
```
Comment (andrych) [inline src/modules/spaces/controllers/spaces.controller.ts:98]: Hoeft deze hier gereassigned te worden?

File: src/modules/assets/services/assets.service.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
+import path from 'path';
+
+import { Injectable, InternalServerErrorException, Logger } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Cron } from '@nestjs/schedule';
+import AWS, { AWSError, S3 } from 'aws-sdk';
+import fse from 'fs-extra';
+import got, { Got } from 'got';
+import _, { kebabCase } from 'lodash';
```
Comment (andrych) [inline src/modules/assets/services/assets.service.ts:9]: Waarom import je `kebabCase` hier nog apart, zit die niet mee onder `_.kebabCase`?

**Review by andrych — APPROVED** [review]

File: src/modules/assets/services/assets.service.ts (hunk 2 above)
Comment (bartnaessens) [reply src/modules/assets/services/assets.service.ts:9]: auto-import denk ik. Fixed!

File: src/modules/spaces/controllers/spaces.controller.ts (hunk 1 above)
Comment (bartnaessens) [reply src/modules/spaces/controllers/spaces.controller.ts:98]: 👌  cleaned it

**Review by bertyhell — APPROVED** [review]

File: src/config/index.ts [hunk 3]
```diff
@@ -90,6 +91,20 @@ const config = (): Configuration => {
...
+		assetServerTokenEndpoint: getEnvValue('ASSET_SERVER_TOKEN_ENDPOINT', true),
+		assetServerTokenSecret: getEnvValue('ASSET_SERVER_TOKEN_SECRET', true),
+		assetServerTokenPassword: getEnvValue('ASSET_SERVER_TOKEN_PASSWORD', true),
+		assetServerTokenUsername: getEnvValue('ASSET_SERVER_TOKEN_USERNAME', true),
+		assetServerBucketName: getEnvValue('ASSET_SERVER_BUCKET_NAME', true),
+		tempAssetFolder: getEnvValue('TEMP_ASSET_FOLDER', false) || '/tmp',
+		multerOptions: {
+			dest: getEnvValue('TEMP_ASSET_FOLDER', false) || '/tmp',
+			limits: {
+				fileSize: 200000000,
```
[the remark this answers is not in the document]
Comment (bartnaessens) [reply src/config/index.ts:104]: Copied from avo 🤷

**Pull request merged**